Source/WebKit/NetworkProcess/cocoa/NetworkTaskCocoa.mm

```diff
@@ -102,6 +103,19 @@ static bool shouldCapCookieExpiryForThirdPartyIPAddress(const WebCore::IPAddress
     return cappedCookies;
 }
 
+#if HAVE(ALLOW_ONLY_PARTITIONED_COOKIES)
+static NSArray<NSHTTPCookie *> *cookiesBySettingPartition(NSArray<NSHTTPCookie *> *cookies, NSString* partition)
+{
+    auto *partitionedCookies = [NSMutableArray arrayWithCapacity:cookies.count];
+    for (NSHTTPCookie *cookie in cookies) {
+        NSHTTPCookie *partitionedCookie = WebCore::NetworkStorageSession::setCookiePartition(cookie, partition);
+        if (partitionedCookie)
+            [partitionedCookies addObject:partitionedCookie];
```

Comment:
It seems partitioned cookie cannot be nil after `setCookiePartition` (unless `cookie` is nil, and if `cookie` can be nil we probably need null check and early return in `setCookiePartition`)